modal/app.py

```diff
@@ -999,12 +1000,26 @@ def cls(
 
         def wrapper(wrapped_cls: Union[CLS_T, _PartialFunction]) -> CLS_T:
             local_state = self._local_state
+            experimental_options_ = experimental_options or {}
+            http_config_ = None
             # Check if the decorated object is a class
             if isinstance(wrapped_cls, _PartialFunction):
                 wrapped_cls.registered = True
                 user_cls = wrapped_cls.user_cls
+                if wrapped_cls.flags & _PartialFunctionFlags.HTTP_WEB_INTERFACE:
+                    http_config = wrapped_cls.params.http_config
+                    if http_config:
+                        flash_region = http_config.proxy_region
+                        experimental_options_["flash"] = flash_region
```

Comment:
Are we still going to be communicating to the server via `experimental_options`? Trying to understand how load bearing this is. The intention with that would be that it's used for "server-only" logic.